### DISCUSSION TITLE
Select which (multiple) pictures to download

### DISCUSSION THREAD

**Discussion by CacklingCapybara** [alangrainger/immich-public-proxy#31] [Feature Requests]
Would be great to be able to multi-select which photos to download, in addition to the single photo download and the "download all" zip option. Eg, when selecting photos, there could be an option to select all, then download the selection.  

I do not know how much complexity this would add, but I do appreciate trying to keep it very lean.

**Comment by morejames** [comment]
@KneelStar  Can you explain how to get the download all button set up with this project or would I have to move over to the docker-immich-share-proxy project to get it?

**Reply by alangrainger** [reply]
@morejames you can enable this in the options by setting `allowDownloadAll` to `true`: 

https://github.com/alangrainger/immich-public-proxy?tab=readme-ov-file#additional-configuration

**Reply by GetPsyched** [reply]
@alangrainger, since the docs updated 3 days ago which makes `allowDownloadAll` to be an `int` type, can you tell us what the default is? I hope the default is `1`, which is to respect Immich's settings.

(also mentioning the defaults for each setting in the doc would be nice)

**Reply by GetPsyched** [reply]
Ah, the change from `bool` to `int` happened a while back in 0bbe409. My bad.

My question still stands, is the default `1`?

**Reply by alangrainger** [reply]
Defaults are here: https://github.com/alangrainger/immich-public-proxy/blob/main/app/config.json

It is `0` in this case. IPP in general uses the least permissive ("safest") defaults, e.g. don't allow downloading.

**Reply by GetPsyched** [reply]
Thanks for the prompt response!

**Reply by GetPsyched** [reply]
@alangrainger, what am I expected to see? I updated the config and restarted `immich-public-proxy.service` with the following config and still don't see the download all button or the gallery title.
```
λ cat /nix/store/pxi1rn2vnzqdxvl5w9gvxhx2v1dlk6pm-config.json
{
  "allowDownloadAll": 1,
  "showGalleryTitle": true
}
```
This JSON is part of `immich-public-proxy.service`:
```toml
[Service]
Environment="IPP_CONFIG=/nix/store/pxi1rn2vnzqdxvl5w9gvxhx2v1dlk6pm-config.json"
...
```

**Reply by alangrainger** [reply]
@GetPsyched go into the shell of the container (`/bin/sh`) and check that you can see your correct data in `/app/config.json`. If not, you haven't mounted the volume correctly for the container to see it.

**Reply by GetPsyched** [reply]
I'm not using containers. The settings are configured simply by the `IPP_CONFIG` environment variable in the systemd service snippet I sent above. This is defined [here](https://github.com/NixOS/nixpkgs/blob/fc02ee70efb805d3b2865908a13ddd4474557ecf/nixos/modules/services/web-apps/immich-public-proxy.nix#L57). Is setting the settings via that variable not supported?

**Reply by alangrainger** [reply]
I don't know anything about NixOS sorry. You might need to ask the question to the maintainer of your linked package.

**Comment by klaus1k** [comment]
Could this be added to IPP or is this a feature, that should be added to lightgallery?

**Reply by alangrainger** [reply]
I'll need to replace lightGallery with another gallery package which supports image selection, but I haven't found a good one yet.

**Reply by klaus1k** [reply]
Thanks, that indeed sounds pretty involved, lightGallery is pretty nice for usability.

**Comment by alangrainger** [comment]
This has been live for a while but I forgot to close the issue
